### ISSUE TITLE
Refactor Type/Environmental Interest

### ISSUE THREAD

**Issue by dougwaldron** [gaepdit/FMS#106]
Based on meeting with program, existing dropdowns may not reflect desired functionality. 

Current "environmental interest" data is obsolete. Only "facility type" is needed and can be renamed "Type/Environmental Interest". Program is reviewing recommendation for final list and migration strategy.

Tasks:
- [x] Refactor entities
- [x] Add EF migration
- [x] Refactor DTOs
- [x] Repositories
- [x] Facility search/add/edit
- ~~Env Interest list editor~~ (part of #87)
- ~~Update data migration scripts~~ (moved to #125)

**Comment by dougwaldron** [comment]
Dropdowns in current application (not necessarily preferable):
![image](https://user-images.githubusercontent.com/737326/96292919-f217a600-0fb7-11eb-8cf7-27507da19ac1.png)
![image](https://user-images.githubusercontent.com/737326/96292936-f643c380-0fb7-11eb-890a-a4be3687b720.png)

**Comment by dougwaldron** [comment]
Current type counts:

```sql
select type, count(*) as [count]
from hwmb.facilityfile
group by type
```

| type   | count |
|--------|-------|
| FUDS   | 21    |
| npl    | 10    |
| hsra   | 862   |
| vrp    | 154   |
| gen    | 5959  |
| brown  | 1121  |
| nongen | 207   |

All should be migrated.

**Comment by dougwaldron** [comment]
Current migration:

```sql
select t.Name, count(*) as [count]
from Facilities f
left join FacilityTypes t
on f.FacilityTypeId = t.Id
group by t.Name
```

| Name   | count |
|--------|-------|
| TSD    | 1     |
| `null` | 196   |
| GEN    | 5648  |
| BROWN  | 805   |
| VRP    | 116   |
| FUDS   | 23    |
| HSRA   | 850   |
| NPL    | 10    |

**Comment by dougwaldron** [comment]
Finer grained:

```sql
select lower(program) as [Env Int], lower(type) as [Facility Type], count(*) as [count]
from hwmb.facilityfile
group by lower(type), lower(program)
order by 1, 2
```

| Env Int | Facility Type | count |
|---------|---------------|-------|
| brown   | brown         | 1116  |
| cercla  | npl           | 10    |
| dod     | fuds          | 21    |
| dod     | nongen        | 12    |
| hsra    | brown         | 5     |
| hsra    | gen           | 7     |
| hsra    | hsra          | 862   |
| hsra    | vrp           | 154   |
| rcra    | gen           | 5952  |
| rcra    | nongen        | 195   |

**Comment by dougwaldron** [comment]
Finer grained:

```sql
select lower(program) as [Env Int], lower(type) as [Facility Type], count(*) as [count]
from hwmb.facilityfile
group by lower(type), lower(program)
order by 1, 2
```

| Env Int | Facility Type | count |
|---------|---------------|-------|
| brown   | brown         | 1116  |
| cercla  | npl           | 10    |
| dod     | fuds          | 21    |
| dod     | nongen        | 12    |
| hsra    | brown         | 5     |
| hsra    | gen           | 7     |
| hsra    | hsra          | 862   |
| hsra    | vrp           | 154   |
| rcra    | gen           | 5952  |
| rcra    | nongen        | 195   |

**Comment by dougwaldron** [comment]
Migration mapping from program:

| Program | Old Type | New Facility Type |
|---------|----------|-------------------|
| any     | brown    | BROWN             |
| any     | vrp      | VRP               |
| any     | hsra     | HSI               |
| any     | gen      | GEN               |
| rcra    | nongen   | TSD/CA            |
| any     | NPL      | NPL               |
| any     | FUDS     | FUDS              |
| dod     | nongen   | DOD               |

**Comment by dougwaldron** [comment]
Closed in #124
